api/v1alpha1/warehouse_types.go

```diff
@@ -306,6 +306,8 @@ type GitSubscription struct {
 }
 
 // ImageSubscription defines a subscription to an image repository.
+//
+// +kubebuilder:validation:XValidation:message="ImageSubscription cannot have both semverConstraint and constraint set",rule="!(has(self.semverConstraint) && has(self.constraint))"
```

Comment:
I feel as if this would be the more standard language for expressing this:

```suggestion
// +kubebuilder:validation:XValidation:message="semverConstraint and constraint fields are mutually exclusive",rule="!(has(self.semverConstraint) && has(self.constraint))"
```

However, my experience with CEL is low up to this point, so cc @hiddeco because idk what best practices we should observe here. (For instance, as with godocs, is it highly encouraged that the first word of the error is exactly the name of the type?)

Reply:
There is no real guidance on what language to use, but this is what I observed being used in other well-known projects in the space.